### PR TITLE
keras_nlp template 

### PR DESCRIPTION
**Pull request by walteriviera**
Just workspace no jupyter

### CODE REVIEW

**Comment by openfl-helper** [comment]
Can one of the admins verify this patch?

**Comment by alexey-gruzdev** [comment]
Jenkins please retry a build

**Comment by alexey-gruzdev** [comment]
Jenkins please retry a build

**Comment by alexey-gruzdev** [comment]
Jenkins please retry a build

File: openfl-workspace/keras_nlp/code/dataloader_utils.py [hunk 1]
```diff
@@ -0,0 +1,213 @@
...
+
+    workspace_dir = getcwd()
+    default_path = path.join(workspace_dir, data_dir)
+    pkgpath = path.join(default_path, pkg)       # path to downloaded zipfile
+    filepath = path.join(default_path, filename)  # path to extracted file
+
+    if path.isfile(filepath):
+        return path.join(data_dir, filename)
+    try:
+        subprocess.run(["wget", url, "-P", default_path], check=True)
```
Comment (psfoley) [inline openfl-workspace/keras_nlp/code/dataloader_utils.py:36]: @walteriviera instead of calling `wget` directly through the shell, could you call `urlretrieve` from urllib as in [this example](https://github.com/intel/openfl/blob/develop/openfl-workspace/torch_cnn_histology/code/histology_utils.py#L39)?

Comment (psfoley) [reply openfl-workspace/keras_nlp/code/dataloader_utils.py:36]: Also after downloading the file, it's integrity should be checked: https://github.com/intel/openfl/blob/develop/openfl-workspace/torch_cnn_histology/code/histology_utils.py#L40-L41

File: openfl-workspace/keras_nlp/code/dataloader_utils.py [hunk 2]
```diff
@@ -0,0 +1,213 @@
...
+    """Create train/val.
+
+    Returns:
+      dict: Results, containing the train-valid split of the dataset (split_ratio = 0.2)
+    """
+    import random
+
+    split_ratio = 0.2
+
+    train_indexes = random.sample(range(num_samples), int(num_samples * (1 - split_ratio)))
```
Comment (psfoley) [inline openfl-workspace/keras_nlp/code/dataloader_utils.py:152]: You should set the seed prior to this split to make the collaborator shards non-overlapping

File: openfl-workspace/keras_nlp/code/dataloader_utils.py (hunk 1 above)
Comment (walteriviera) [reply openfl-workspace/keras_nlp/code/dataloader_utils.py]: Fixed. Thanks Patrick 👍

File: openfl-workspace/keras_nlp/code/dataloader_utils.py (hunk 2 above)
Comment (walteriviera) [reply openfl-workspace/keras_nlp/code/dataloader_utils.py]: Done, thanks. 👍

**Comment by walteriviera** [comment]
I have read the CLA Document and I hereby sign the CLA

**Comment by walteriviera** [comment]
I have read the CLA Document and I hereby sign the CLA

File: openfl-workspace/keras_nlp/code/dataloader_utils.py [hunk 3]
```diff
@@ -0,0 +1,231 @@
...
+                  Max sequence length for outputs = {max_decoder_seq_length}")
+
+    logger.info(f"[DL]-import_raw_data: encoder_input_data = {encoder_input_data.shape}")
+    logger.info(f"[DL]-import_raw_data: decoder_input_data = {decoder_input_data.shape}")
+    logger.info(f"[DL]-import_raw_data: decoder_target_data = {decoder_target_data.shape}")
+
+    return details, encoder_input_data, decoder_input_data, decoder_target_data
+
+
+<<<<<<< HEAD
```
Comment (psfoley) [inline openfl-workspace/keras_nlp/code/dataloader_utils.py:148]: It looks like there may have been some conflicts from your local merge here. This should be cleaned up

File: openfl-workspace/keras_nlp/code/dataloader_utils.py [hunk 4]
```diff
@@ -0,0 +1,231 @@
...
+def get_datasets_(encoder_input_data, decoder_input_data, decoder_target_data, num_samples, split_ratio):
+=======
```
Comment (psfoley) [inline openfl-workspace/keras_nlp/code/dataloader_utils.py:150]: Cleanup from merge

File: openfl-workspace/keras_nlp/code/dataloader_utils.py [hunk 5]
```diff
@@ -0,0 +1,231 @@
...
+    Returns:
+      dict: Results, containing the train-valid split of the dataset (split_ratio = 0.2)
+    """
+<<<<<<< HEAD
+    import random
+    
+    random.seed(42)
+    train_indexes = random.sample(range(num_samples), int(num_samples * (1 - split_ratio)))
+=======
+>>>>>>> a041924... pythonized download fn + set split ratio and num_samples to params
```
Comment (alexey-gruzdev) [inline openfl-workspace/keras_nlp/code/dataloader_utils.py:164]: The same as @psfoley mentioned. @walteriviera could you re-check all changes for the artifacts from merge.

Comment (walteriviera) [reply openfl-workspace/keras_nlp/code/dataloader_utils.py]: Done. Thanks
